mono/metadata/threadpool.c

```diff
@@ -692,12 +613,18 @@ ves_icall_System_Threading_ThreadPool_GetAvailableThreadsNative (gint32 *worker_
 	if (!worker_threads || !completion_port_threads)
 		return;
 
-	mono_lazy_initialize (&status, initialize);
+	if (!mono_lazy_initialize (&status, initialize) || !mono_refcount_tryinc (&threadpool)) {
```

Comment:
This doesn't look good.

We have the same information (subsystem lifecycle state) encoded in two variables. And they are accessed in a lock-free way.